src/workers/build/mod.rs

```diff
@@ -0,0 +1,151 @@
+    // add other resoureces
+    Ok(worker)
+}
+
+fn wasm_worker(project_name: &str) -> Result<Worker, failure::Error> {
+    let name = krate::Krate::new("./")?.name.replace("-", "_");
+
+    build_generated_dir()?;
+    concat_js(&name)?;
+
+    let script_path = "./worker/generated/script.js";
+    let wasm_path = format!("./pkg/{}_bg.wasm", name);
+
+    let wasm_resource = wasm_resource(wasm_path)?;
+
+    Ok(Worker {
+        name: project_name.to_string(),
+        script: Script {
+            name: "script".to_string(),
+            path: script_path.to_string(),
+        },
+        resources: vec![wasm_resource],
+    })
+}
+
+fn webpack_worker(project_name: &str) -> Result<Worker, failure::Error> {
+    let script_path = "./worker/script.js";
+    let mut worker = Worker {
+        name: project_name.to_string(),
+        script: Script {
+            name: "script".to_string(),
+            path: script_path.to_string(),
+        },
+        resources: Vec::new(),
+    };
+    let wasm_path = "./worker/module.wasm";
+    if Path::new(wasm_path).exists() {
+        let wasm_resource = wasm_resource(wasm_path.to_string())?;
+        worker.resources.push(wasm_resource);
+    };
+    Ok(worker)
+}
+
+fn js_worker(project_name: &str) -> Result<Worker, failure::Error> {
+    let pkg = Package::new("./")?;
+    let script_path = pkg.main()?;
+    Ok(Worker {
+        name: project_name.to_string(),
+        script: Script {
+            name: "script".to_string(),
+            path: script_path,
+        },
+        resources: Vec::new(),
+    })
+}
+
+fn wasm_resource(wasm_path: String) -> Result<Resource, failure::Error> {
+    let metadata_path = "./worker/metadata_wasm.json";
+    let mut binding = "wasm".to_string();
+
+    if Path::new(metadata_path).exists() {
+        let file = fs::File::open(metadata_path)?;
+        let reader = BufReader::new(file);
+        let data: serde_json::Value = serde_json::from_reader(reader)?;
+        binding = match data["bindings"][0]["name"].as_str() {
+            Some(s) => s.to_string(),
+            None => {
+                failure::bail!("binding was not a string");
+            }
+        };
+    }
+    Ok(Resource::WasmModule(WasmModule {
+        path: wasm_path.to_string(),
+        binding: binding,
+    }))
+}
+
+fn concat_js(name: &str) -> Result<(), failure::Error> {
+    let bindgen_js_path = format!("./pkg/{}.js", name);
+    let bindgen_js: String = fs::read_to_string(bindgen_js_path)?.parse()?;
+
+    let worker_js: String = fs::read_to_string("./worker/worker.js")?.parse()?;
+    let js = format!("{} {}", bindgen_js, worker_js);
+
+    fs::write("./worker/generated/script.js", js.as_bytes())?;
+    Ok(())
+}
+
+fn build_generated_dir() -> Result<(), failure::Error> {
+    let dir = "./worker/generated";
+    if !Path::new(dir).is_dir() {
+        fs::create_dir("./worker/generated")?;
+    }
+    Ok(())
+}
+
+fn command(args: &[&str], binary_path: PathBuf) -> Command {
```

Comment:
don't fully understand why this is here,as it relates to how wrangler calls commands

Reply:
It was in the build command module, but was moved here as it's more or less business logic for building workers and shouldn't be in a command module. It's just here because this is the only file that references it. Do you have a preferred location?

Reply:
let's create a top-level `utils` folder (not my favorite convention but appropriate here) and export it from there.